src/SignalRServiceExtension/Bindings/SignalRInputBindings/SignalRValueProvider.cs

```diff
@@ -10,10 +10,14 @@ namespace Microsoft.Azure.WebJobs.Extensions.SignalRService
     internal class SignalRValueProvider : IValueProvider
     {
         private object value;
+        private string invokeString;
 
-        public SignalRValueProvider(object value)
+        // todo: fix invoke string in another PR
+        public SignalRValueProvider(object value, Type type, string invokeString)
```

Comment:
>string invokeString [](start = 61, length = 19)

Why add type/invokeString into parameter? couldn't understand its usage. JS is blocked in InputBindingProvider attribute resolver, anything related to here?

Reply:
The js require the `IValueProvider` returned by `IBinding` has `Type` property that is a `string`. It should be the concrete type of the value of the `IValueProvider`. The attribute resolver is used for extract the final value of the binding expression, e.g. get a real user ID from http request: UserId="{req.header.x-client-userId}"

Reply:
I see.

---
In reply to: [411085354](https://github.com/Azure/azure-functions-signalrservice-extension/pull/119#discussion_r411085354) [](ancestors = 411085354)